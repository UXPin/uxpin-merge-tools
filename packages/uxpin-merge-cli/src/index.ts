import 'source-map-support/register';
export { runProgram } from './program/runProgram';
export { Command } from './program/command/Command';
export { getToolVersion } from './program/utils/version/getToolVersion';
export { buildDesignSystemWithStorybook } from './steps/building/buildDesignSystemWithStorybook';
