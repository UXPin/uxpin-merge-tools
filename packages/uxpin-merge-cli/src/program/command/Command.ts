export enum Command {
  DUMP = 'dump',
  EXPERIMENT = 'experiment',
  INIT = 'init',
  GENERATE_PRESETS = 'generate-presets',
  PUSH = 'push',
  DELETE_VERSION = 'delete-version',
  SERVER = 'server',
  SUMMARY = 'summary',
}

export const DEFAULT_COMMAND: Command = Command.EXPERIMENT;
