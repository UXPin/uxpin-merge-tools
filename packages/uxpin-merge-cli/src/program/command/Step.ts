import { DSMetadata } from '../DSMeta';

export interface Step {
  shouldRun: boolean;
  exec: (infos: DSMetadata) => any;
}

export type StepExecutor = (designSystem: DSMetadata) => Promise<DSMetadata>;

export type StepWithoutDSExecutor = () => Promise<void>;
