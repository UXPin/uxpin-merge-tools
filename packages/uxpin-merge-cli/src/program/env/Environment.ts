export enum Environment {
  DEVELOPMENT = 'development',
  TEST = 'test',
  PRODUCTION = 'production',
}

export const DEFAULT_ENVIRONMENT: Environment = Environment.PRODUCTION;
