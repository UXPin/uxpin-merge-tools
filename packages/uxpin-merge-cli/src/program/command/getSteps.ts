import { ProgramArgs } from '../args/ProgramArgs';
import { Command } from './Command';
import { getDeleteVersionCommandSteps } from './delete_version/getDeleteVersionCommandSteps';
import { getDumpCommandSteps } from './dump/getDumpCommandSteps';
import { getExperimentationCommandSteps } from './experimentation/getExperimentationCommandSteps';
import { getGeneratePresetsCommandSteps } from './generate_presets/getGeneratePresetsCommandSteps';
import { getInitCommandSteps } from './init/getInitCommandSteps';
import { getPushCommandSteps } from './push/getPushCommandSteps';
import { getServerCommandSteps } from './server/getServerCommandSteps';
import { Step } from './Step';
import { getSummaryCommandSteps } from './summary/getSummaryCommandSteps';

export function getSteps(args: ProgramArgs): Step[] {
  switch (args.command) {
    case Command.DUMP:
      return getDumpCommandSteps();

    case Command.EXPERIMENT:
      return getExperimentationCommandSteps(args);

    case Command.INIT:
      return getInitCommandSteps(args);

    case Command.GENERATE_PRESETS:
      return getGeneratePresetsCommandSteps(args);

    case Command.PUSH:
      return getPushCommandSteps(args);

    case Command.DELETE_VERSION:
      return getDeleteVersionCommandSteps(args);

    case Command.SERVER:
      return getServerCommandSteps(args);

    case Command.SUMMARY:
      return getSummaryCommandSteps();

    default:
      return [];
  }
}
