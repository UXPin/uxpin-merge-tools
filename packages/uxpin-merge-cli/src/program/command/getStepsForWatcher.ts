import { ProgramArgs } from '../args/ProgramArgs';
import { Command } from './Command';
import { getExperimentationWatchCommandSteps } from './experimentation/getExperimentationWatchCommandSteps';
import { Step } from './Step';

export function getStepsForWatcher(args: ProgramArgs): Step[] {
  if (args.command === Command.EXPERIMENT) {
    return getExperimentationWatchCommandSteps(args);
  }

  return [];
}
