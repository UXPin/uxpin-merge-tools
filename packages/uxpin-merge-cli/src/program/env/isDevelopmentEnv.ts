import { Environment } from './Environment';

export function isDevelopmentEnv(): boolean {
  return process.env.UXPIN_ENV === Environment.DEVELOPMENT;
}
