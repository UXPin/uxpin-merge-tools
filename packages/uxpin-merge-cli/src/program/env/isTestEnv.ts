import { Environment } from './Environment';

export function isTestEnv(): boolean {
  return process.env.UXPIN_ENV === Environment.TEST;
}
