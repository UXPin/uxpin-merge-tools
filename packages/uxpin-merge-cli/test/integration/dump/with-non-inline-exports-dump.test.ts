import { Command } from '../../../src';
import { Environment } from '../../../src/program/env/Environment';
import { emptyLatestCommitStub } from '../../resources/stubs/emptyLatestCommit';
import { runUXPinMergeCommand } from '../../utils/command/runUXPinMergeCommand';
import { setTimeoutBeforeAll } from '../../utils/command/setTimeoutBeforeAll';
import { setupStubbyServer } from '../../utils/stubby/setupStubbyServer';

const CURRENT_TIMEOUT = 30000;
setTimeoutBeforeAll(CURRENT_TIMEOUT);

jest.mock('../../../src/program/utils/version/getToolVersion');

describe('The dump command', () => {
  const { getTlsPort } = setupStubbyServer(emptyLatestCommitStub);

  describe('run for the with-non-inline-exports repository', () => {
    it('prints components with proper properties', () => {
      // when
      return runUXPinMergeCommand({
        cwd: 'resources/designSystems/withNonInlineExports',
        env: {
          UXPIN_API_DOMAIN: `0.0.0.0:${getTlsPort()}`,
          UXPIN_ENV: Environment.TEST,
        },
        params: [Command.DUMP, '--config="./uxpin.config.js"'],
      }).then((consoleOutput) => {
        // then
        const output: any = JSON.parse(consoleOutput);
        expect(output.categorizedComponents).toMatchSnapshot();
      });
    });
  });
});
