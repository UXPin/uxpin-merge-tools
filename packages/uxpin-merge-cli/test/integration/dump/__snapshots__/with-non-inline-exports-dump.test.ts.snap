// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The dump command run for the with-non-inline-exports repository prints components with proper properties 1`] = `
[
  {
    "components": [
      {
        "defaultExported": false,
        "documentation": {
          "examples": [],
        },
        "info": {
          "dirPath": "src/components/Avatar",
          "implementation": {
            "framework": "reactjs",
            "lang": "typescript",
            "path": "src/components/Avatar/Avatar.tsx",
          },
          "presets": [],
        },
        "name": "Avatar",
        "presets": [],
        "properties": [
          {
            "description": "",
            "isRequired": true,
            "name": "size",
            "type": {
              "name": "string",
              "structure": {},
            },
          },
          {
            "description": "",
            "isRequired": true,
            "name": "imageUrl",
            "type": {
              "name": "string",
              "structure": {},
            },
          },
        ],
        "wrappers": [],
      },
      {
        "defaultExported": false,
        "documentation": {
          "examples": [],
        },
        "info": {
          "dirPath": "src/components/AvatarFn",
          "implementation": {
            "framework": "reactjs",
            "lang": "typescript",
            "path": "src/components/AvatarFn/AvatarFn.tsx",
          },
          "presets": [],
        },
        "name": "AvatarFn",
        "presets": [],
        "properties": [
          {
            "description": "",
            "isRequired": true,
            "name": "size",
            "type": {
              "name": "string",
              "structure": {},
            },
          },
          {
            "description": "",
            "isRequired": true,
            "name": "imageUrl",
            "type": {
              "name": "string",
              "structure": {},
            },
          },
        ],
        "wrappers": [],
      },
      {
        "defaultExported": true,
        "documentation": {
          "examples": [],
        },
        "info": {
          "dirPath": "src/components/AvatarFnDefault",
          "implementation": {
            "framework": "reactjs",
            "lang": "typescript",
            "path": "src/components/AvatarFnDefault/AvatarFnDefault.tsx",
          },
          "presets": [],
        },
        "name": "AvatarFnDefault",
        "presets": [],
        "properties": [
          {
            "description": "",
            "isRequired": true,
            "name": "size",
            "type": {
              "name": "string",
              "structure": {},
            },
          },
          {
            "description": "",
            "isRequired": true,
            "name": "imageUrl",
            "type": {
              "name": "string",
              "structure": {},
            },
          },
        ],
        "wrappers": [],
      },
      {
        "defaultExported": true,
        "documentation": {
          "examples": [],
        },
        "info": {
          "dirPath": "src/components/AvatarHOC",
          "implementation": {
            "framework": "reactjs",
            "lang": "typescript",
            "path": "src/components/AvatarHOC/AvatarHOC.tsx",
          },
          "presets": [],
        },
        "name": "AvatarHOC",
        "presets": [],
        "properties": [
          {
            "description": "",
            "isRequired": true,
            "name": "size",
            "type": {
              "name": "string",
              "structure": {},
            },
          },
          {
            "description": "",
            "isRequired": true,
            "name": "imageUrl",
            "type": {
              "name": "string",
              "structure": {},
            },
          },
        ],
        "wrappers": [],
      },
      {
        "defaultExported": true,
        "documentation": {
          "examples": [],
        },
        "info": {
          "dirPath": "src/components/ButtonWithIconAsProp",
          "implementation": {
            "framework": "reactjs",
            "lang": "typescript",
            "path": "src/components/ButtonWithIconAsProp/ButtonWithIconAsProp.tsx",
          },
          "presets": [],
        },
        "name": "ButtonWithIconAsProp",
        "presets": [],
        "properties": [
          {
            "description": "",
            "isRequired": true,
            "name": "children",
            "type": {
              "name": "node",
              "structure": {},
            },
          },
          {
            "description": "",
            "isRequired": false,
            "name": "isPrimary",
            "type": {
              "name": "boolean",
              "structure": {},
            },
          },
        ],
        "wrappers": [],
      },
      {
        "defaultExported": true,
        "documentation": {
          "examples": [],
        },
        "info": {
          "dirPath": "src/components/MenuWithData",
          "implementation": {
            "framework": "reactjs",
            "lang": "typescript",
            "path": "src/components/MenuWithData/MenuWithData.tsx",
          },
          "presets": [],
        },
        "name": "MenuWithData",
        "presets": [],
        "properties": [
          {
            "description": "",
            "isRequired": true,
            "name": "data",
            "type": {
              "name": "array",
              "structure": {},
            },
          },
        ],
        "wrappers": [],
      },
      {
        "defaultExported": true,
        "documentation": {
          "examples": [],
        },
        "info": {
          "dirPath": "src/components/MenuWithDataHOC",
          "implementation": {
            "framework": "reactjs",
            "lang": "typescript",
            "path": "src/components/MenuWithDataHOC/MenuWithDataHOC.tsx",
          },
          "presets": [],
        },
        "name": "MenuWithDataHOC",
        "presets": [],
        "properties": [
          {
            "description": "",
            "isRequired": true,
            "name": "data",
            "type": {
              "name": "array",
              "structure": {},
            },
          },
        ],
        "wrappers": [],
      },
    ],
    "name": "General",
  },
]
`;
